tests/dualtor/test_server_failure.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+import time
+from tests.common.dualtor.mux_simulator_control import toggle_simulator_port_to_upper_tor, simulator_flap_counter, simulator_server_down
+from tests.common.helpers.assertions import pytest_assert, pytest_require
+from tests.common.dualtor.dual_tor_utils import show_muxcable_status, rand_selected_interface
+from tests.common.fixtures.ptfhost_utils import change_mac_addresses, run_garp_service, run_icmp_responder
+
+pytestmark = [
+    pytest.mark.topology('t0'),
+    pytest.mark.usefixtures('run_garp_service', 'run_icmp_responder')
+]
+
+
+def test_server_down(duthosts, tbinfo, rand_selected_interface, simulator_flap_counter, simulator_server_down, toggle_simulator_port_to_upper_tor, loganalyzer):
+    """
+    Verify that mux cable is not toggled excessively.
+    """
+    pytest_require('dualtor' in tbinfo['topo']['name'], 
+                    "Only run on dualtor testbed")
+
+    for analyzer in list(loganalyzer.values()):
+        analyzer.ignore_regex.append(r".*ERR swss#orchagent: :- setState: State transition from active to active is not-handled")
+        
+    upper_tor = duthosts[tbinfo['duts'][0]]
+    lower_tor = duthosts[tbinfo['duts'][1]]
+    PAUSE_TIME = 5
+
+    itfs, _ = rand_selected_interface
+    # Set upper_tor as active
+    toggle_simulator_port_to_upper_tor(itfs)
+    time.sleep(PAUSE_TIME)
+    mux_flap_counter_0 = simulator_flap_counter(itfs)
+    # Server down
+    simulator_server_down(itfs)
```

Comment:
Instead of using this method please stop the ICMP responder on the PTF as that is more similar to the real-life scenario we are testing for.

Reply:
I was thinking stopping ICMP responder to simulate this scenario. But there is a limitation in ```icmp_responder```, that is we are not able to stop hearbeat from a single server. So stopping ```icmp_responder``` is killing all servers.

Reply:
This is a good point, thanks!